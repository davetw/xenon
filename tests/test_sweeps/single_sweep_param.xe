begin sweep sweep1

use xenon.tests.machsuite

generate configs

# Set parameters.
set output_dir "test" # A comment
set source_dir "test"
set simulator "aladdin"
set memory_type "spad"

# Only sweep this variable.
sweep cycle_time from 1 to 5

end sweep
