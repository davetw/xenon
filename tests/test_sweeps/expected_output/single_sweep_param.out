{ 'Benchmark("aes-aes")': { 'Array("buf")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("ctx")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("k")': { 'partition_factor': 1,
                                            'partition_type': 'cyclic'},
                            'Array("rcon")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Array("sbox")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Function("aes256_encrypt_ecb")': { 'Loop("ecb1")': { 'unrolling': 1},
                                                                'Loop("ecb2")': { 'unrolling': 1},
                                                                'Loop("ecb3")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey")': { 'Loop("addkey")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey_cpy")': { 'Loop("cpkey")': { 'unrolling': 1}},
                            'Function("aes_subBytes")': { 'Loop("sub")': { 'unrolling': 1}},
                            'cycle_time': 1,
                            'pipelining': False}}
{ 'Benchmark("aes-aes")': { 'Array("buf")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("ctx")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("k")': { 'partition_factor': 1,
                                            'partition_type': 'cyclic'},
                            'Array("rcon")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Array("sbox")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Function("aes256_encrypt_ecb")': { 'Loop("ecb1")': { 'unrolling': 1},
                                                                'Loop("ecb2")': { 'unrolling': 1},
                                                                'Loop("ecb3")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey")': { 'Loop("addkey")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey_cpy")': { 'Loop("cpkey")': { 'unrolling': 1}},
                            'Function("aes_subBytes")': { 'Loop("sub")': { 'unrolling': 1}},
                            'cycle_time': 2,
                            'pipelining': False}}
{ 'Benchmark("aes-aes")': { 'Array("buf")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("ctx")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("k")': { 'partition_factor': 1,
                                            'partition_type': 'cyclic'},
                            'Array("rcon")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Array("sbox")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Function("aes256_encrypt_ecb")': { 'Loop("ecb1")': { 'unrolling': 1},
                                                                'Loop("ecb2")': { 'unrolling': 1},
                                                                'Loop("ecb3")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey")': { 'Loop("addkey")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey_cpy")': { 'Loop("cpkey")': { 'unrolling': 1}},
                            'Function("aes_subBytes")': { 'Loop("sub")': { 'unrolling': 1}},
                            'cycle_time': 3,
                            'pipelining': False}}
{ 'Benchmark("aes-aes")': { 'Array("buf")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("ctx")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("k")': { 'partition_factor': 1,
                                            'partition_type': 'cyclic'},
                            'Array("rcon")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Array("sbox")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Function("aes256_encrypt_ecb")': { 'Loop("ecb1")': { 'unrolling': 1},
                                                                'Loop("ecb2")': { 'unrolling': 1},
                                                                'Loop("ecb3")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey")': { 'Loop("addkey")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey_cpy")': { 'Loop("cpkey")': { 'unrolling': 1}},
                            'Function("aes_subBytes")': { 'Loop("sub")': { 'unrolling': 1}},
                            'cycle_time': 4,
                            'pipelining': False}}
{ 'Benchmark("aes-aes")': { 'Array("buf")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("ctx")': { 'partition_factor': 1,
                                              'partition_type': 'cyclic'},
                            'Array("k")': { 'partition_factor': 1,
                                            'partition_type': 'cyclic'},
                            'Array("rcon")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Array("sbox")': { 'partition_factor': 1,
                                               'partition_type': 'cyclic'},
                            'Function("aes256_encrypt_ecb")': { 'Loop("ecb1")': { 'unrolling': 1},
                                                                'Loop("ecb2")': { 'unrolling': 1},
                                                                'Loop("ecb3")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey")': { 'Loop("addkey")': { 'unrolling': 1}},
                            'Function("aes_addRoundKey_cpy")': { 'Loop("cpkey")': { 'unrolling': 1}},
                            'Function("aes_subBytes")': { 'Loop("sub")': { 'unrolling': 1}},
                            'cycle_time': 5,
                            'pipelining': False}}
